pkg/kubectl/cmd/expose.go

```diff
@@ -189,21 +189,29 @@ func RunExpose(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
 		}
 	}
 
+	resourceMapper := &resource.Mapper{ObjectTyper: typer, RESTMapper: mapper, ClientMapper: f.ClientMapperForCommand()}
+	info, err = resourceMapper.InfoForObject(object)
+	if err != nil {
+		return err
+	}
 	// TODO: extract this flag to a central location, when such a location exists.
-	if !cmdutil.GetFlagBool(cmd, "dry-run") {
-		resourceMapper := &resource.Mapper{ObjectTyper: typer, RESTMapper: mapper, ClientMapper: f.ClientMapperForCommand()}
-		info, err := resourceMapper.InfoForObject(object)
-		if err != nil {
-			return err
-		}
+	if cmdutil.GetFlagBool(cmd, "dry-run") {
+		fmt.Fprintln(out, "running in dry-run mode...")
+	} else {
 		data, err := info.Mapping.Codec.Encode(object)
 		if err != nil {
 			return err
 		}
-		_, err = resource.NewHelper(info.Client, info.Mapping).Create(namespace, false, data)
+		object, err = resource.NewHelper(info.Client, info.Mapping).Create(namespace, false, data)
 		if err != nil {
 			return err
 		}
 	}
-	return f.PrintObject(cmd, object, out)
+	outputFormat := cmdutil.GetFlagString(cmd, "output")
+	if outputFormat == "" {
+		cmdutil.PrintSuccess(mapper, false, out, info.Mapping.Resource, info.Name, "exposed")
+	} else {
+		return f.PrintObject(cmd, object, out)
```

Comment:
Invert this condition so that the return statement is in the if and then drop the else